Assets/Scripts/Control&Input/ItemSelectManager.cs

```diff
@@ -0,0 +1,300 @@
+    private List<Item> extensionItems;
+    private InputManager inputManager;
+    [SerializeField] 
+    private Selectable defaultSelector;
+    private Vector2 moveInput;
+    private int bodyIndex;
+    private int barrelIndex;
+    private int extensionIndex;
+    [SerializeField]
+    private TMP_Text timerText;
+    private const float errorMarginInput = 0.1f;
+    private bool gamepadMoveReady = true;
+    public Transform cameraPosition;
+    [SerializeField]
+    private PlayerStatUI playerStatUI;
+    [SerializeField]
+    private TMP_Text secretName;
+
+    [SerializeField] 
+    private Image bodySelect;
+    [SerializeField] 
+    private Image barrelSelect;
+    [SerializeField] 
+    private Image extensionSelect; 
+    [SerializeField]
+    private Color selectedColor;
+    [SerializeField]
+    private Color defaultColor;
+    private int selectedIndex = 0;
+
+   void Start()
+    {
+        bodySelect.color = selectedColor;
+    }
+    public IEnumerator SpawnItems(InputManager inputManager){
+
+        //Without this line it is arbitrary whether an item from aucion is transferred
+        yield return null;
+        this.inputManager = inputManager;
+        canvas.worldCamera = inputManager.GetComponent<Camera>();
+        playerStatUI.PlayerIdentity = inputManager.GetComponent<PlayerIdentity>();
+        bodyItems = inputManager.GetComponent<PlayerIdentity>().Bodies;
+        barrelItems = inputManager.GetComponent<PlayerIdentity>().Barrels;
+        extensionItems = inputManager.GetComponent<PlayerIdentity>().Extensions;   
+
+        InstantiateItems(bodyItems,defaultBodyItem,itemSpawnPoints[0], bodies);
+        InstantiateItems(barrelItems,defaultBarrelItem,itemSpawnPoints[1], barrels);
+        InstantiateItems(extensionItems,null,itemSpawnPoints[2],extensions);
+
+       
+        bodyIndex = bodies.Count - 1;
+        barrelIndex = barrels.Count - 1;
+        extensionIndex = extensions.Count - 1;
+        
+
+
+        ChangeItemDisplayed(bodies[bodyIndex], bodies[bodyIndex], itemSpawnPoints[0]);
+        ChangeItemDisplayed(barrels[barrelIndex],barrels[barrelIndex],itemSpawnPoints[1]);
+        ChangeItemDisplayed(
+            extensions.Count != 0 ? extensions[extensionIndex] : null, 
+            extensions.Count != 0 ? extensions[extensionIndex] : null, 
+            itemSpawnPoints[2]);
+        
+        yield return null;
+        yield return null;
+
+        inputManager.onMovePerformed += MoveInputPerformed;
+        inputManager.onMoveCanceled += MoveInputCanceled;
+
+        timer.StartTimer(20f);
+        timer.OnTimerUpdate += UpdateTimer;
+        timer.OnTimerRunCompleted += ChangeScene;
+        timer.OnTimerRunCompleted += SetLoadout;
+        timer.OnTimerRunCompleted -= UpdateTimer;
+
+
+    }
+
+    private void ChangeItemDisplayed(GameObject previousItem, GameObject nextItem,Transform itemSpawnPoint){
+
+        if(previousItem != null && nextItem != null){
```

Comment:
This `if` should be inverted so you return early instead

Reply:
Fixed it!